CDP4WebServices.API/Services/Operations/SideEffects/Implementation/IterationSetupSideEffect.cs

```diff
@@ -203,11 +222,26 @@ public override void AfterCreate(IterationSetup thing, Thing container, Iteratio
         /// </param>
         public override void BeforeDelete(IterationSetup thing, Thing container, NpgsqlTransaction transaction, string partition, ISecurityContext securityContext)
         {
-            if (thing.FrozenOn == null)
+            var iterationSetups = this.IterationSetupService.GetShallow(transaction, partition, new [] {thing.Iid}, securityContext).OfType<IterationSetup>();
+            var iterationSetup = iterationSetups.SingleOrDefault();
+
+            if (iterationSetup == null)
+            {
+                throw new InvalidOperationException("IterationSetup is null.");
+            }
+
+            if (iterationSetup.FrozenOn == null)
             {
                 throw new InvalidOperationException("It is not possible to delete the current iteration.");
             }
 
+            if (iterationSetup.IsDeleted != false)
+            {
+                return;
+            }
+
+            iterationSetup.IsDeleted = true;
```

Comment:
we reach line 243 when the `iterationSetup.IsDeleted = true`, why do we need to set it and then persist that in the dbase?

Reply:
Deleting the contents of an iteration means removing the Iteration and all the objects it contains from the data store, but we have to remain descriptive data in IterationSetup for other objects, so we cannot delete iterationSetup, but only mark it like a deleted. This approach prevent any wrong read of the associated Iteration data (on any level within the Iteration) and prevent showing deleted iterations when requesting.